### ISSUE TITLE
Prompt configuration

### ISSUE THREAD

**Issue by takluyver** [ipython/ipython#9292]
Opening this so I don't forget to look at it for 5.0.

In 4.x, four kinds of prompt are generated by `PromptManager` (one of my early contributions to IPython; ah, nostalgia):

```
In  [1]: Input prompts
    ...: Continuation prompts
-------> Rewriting prompts
Out [1]: Output prompts
```

With the prompt_toolkit changes now merged, the first two of those are no longer made by PromptManager, because prompt_toolkit needs a list of pygments tokens, plus associated style information to colour them. Output prompts and the little-used rewriting prompt are still done through PromptManager for now.

I want to try to bring some coherence to this so people can customise the prompts if they want, and then I should document the interface.

**Comment by Carreau** [comment]
Should/could we share code/syntax/config options names with Xonsh ?

**Comment by Carreau** [comment]
That will be closed by #9471 I guess.

**Comment by Carreau** [comment]
Closed by #9471.
